contracts/proxy/transparent/TransparentUpgradeableProxy.sol

```diff
@@ -70,73 +68,40 @@ contract TransparentUpgradeableProxy is ERC1967Proxy {
     error ProxyDeniedAdminAccess();
 
     /**
-     * @dev msg.value is not 0.
-     */
-    error ProxyNonPayableFunction();
-
-    /**
-     * @dev Initializes an upgradeable proxy managed by `_admin`, backed by the implementation at `_logic`, and
-     * optionally initialized with `_data` as explained in {ERC1967Proxy-constructor}.
+     * @dev Initializes an upgradeable proxy managed by an instance of a {ProxyAdmin} with an `initialOwner`,
+     * backed by the implementation at `_logic`, and optionally initialized with `_data` as explained in
+     * {ERC1967Proxy-constructor}.
      */
-    constructor(address _logic, address admin_, bytes memory _data) payable ERC1967Proxy(_logic, _data) {
-        _admin = admin_;
+    constructor(address _logic, address initialOwner, bytes memory _data) payable ERC1967Proxy(_logic, _data) {
```

Comment:
Hey i just stumbled over this change when reviewing v5 and was wondering if it wouldn't make sense to have an overload with `ProxyAdmin _admin` instead of `address initialOwner`.

This way one could easily reuse a single ProxyAdmin for multiple contracts.

Reply:
> This way one could easily reuse a single ProxyAdmin for multiple contracts.

This used ti be the case in v4. We decided to change that.

In v4, the admin was stored in storage, which means that any call going trough the proxy would have to load that value, costing 2100 gas.

```solidity
function _fallback() internal virtual override {
        if (msg.sender == _admin) {
            ...
```

We wanted to reduce that gas cost, and so we decided to make the admin immutable. That means that it is not possible for a proxy to change which admin it uses. You could change the value stored in the ERC-1967 slot, but that would have no effect.

That means that if multiple proxy use the same ProxyAdmin, then you can separate the ownership. You may be able to change who owns the ProxyAdmin, but every proxy that point to it will share the same owner, forever.

That is why we decided to make the `TransparentUpgradeableProxy <> ProxyAdmin` a 1-to-1 relationship.

Reply:
Agree with @Amxx.

We agreed that it's better to reduce the SLOAD cost in the proxy than allowing to reuse ProxyAdmins. The tradeoff is fine because this not only increases deployment costs but reduces every proxy interaction cost. Consider regular proxy interactions are more frequent (potentially millions of times) than admin interactions.

Reply:
I just stumbled upon this while deploying a proxy using v5. Being used to the v4 behavior, I wrongly passed in a proxy admin making the deployment unusable. As far as I could tell, this change is not documented in any changelog. I think that it would be good to add this to the v5 changelog.

Reply:
Thanks for sharing @danhper! The change is documented on both [OZ upgrades plugins](https://github.com/OpenZeppelin/openzeppelin-upgrades/releases/tag/%40openzeppelin%2Fhardhat-upgrades%403.0.0) and [OZ Contracts](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/CHANGELOG.md#proxy) (see Proxy section). Though I know it's counterintuitive, we strongly advice users to be careful about [major releases](https://docs.openzeppelin.com/contracts/5.x/backwards-compatibility#major_releases).

We'd be happy to hear suggestions on how to improve the way we communicate these changes.